File: Algorithm/subjects/1801-1900/1822. 数组元素积的符号/main.java
class Solution {
    public int arraySign(int[] nums) {
        boolean status = false;
        int positiveNum = 0, negativeNum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                return 0;
            } else if (nums[i] > 0) {
                positiveNum++;
            } else if (nums[i] < 0) {
                negativeNum++;
            }
        }
        if (negativeNum % 2 == 0) {
            return 1;
        } else {
            return -1;
        }
    }
}

File: Algorithm/subjects/1801-1900/1822. 数组元素积的符号/main.cpp
class Solution {
public:
    int arraySign(vector<int>& nums) {
        int n = nums.size();
        int sum = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == 0) {
                return 0;
            }
            if (nums[i] < 0) {
                sum++;
            }
        }
        return sum % 2 == 0 ? 1 : -1;
    }
};